pkgs/by-name/op/openvpn3/package.nix

```diff
@@ -0,0 +1,133 @@
+{ lib
+, stdenv
+, fetchFromGitHub
+, asio
+, glib
+, jsoncpp
+, libcap_ng
+, libnl
+, libuuid
+, lz4
+, openssl
+, pkg-config
+, protobuf
+, python3
+, systemd
+, tinyxml-2
+, wrapGAppsHook3
+, gobject-introspection
+, meson
+, ninja
+, gdbuspp
+, cmake
+, git
+, enableSystemdResolved ? true
+}:
+
+stdenv.mkDerivation rec {
+  pname = "openvpn3";
+  # also update openvpn3-core
+  version = "22_dev";
+
+  src = fetchFromGitHub {
+    owner = "OpenVPN";
+    repo = "openvpn3-linux";
+    rev = "refs/tags/v${version}";
+    # Don't forget to actualize version scripts in `postPatch`
+    hash = "sha256-UbphN5gHgO30ry0kX6W4qSs5Ksrvfhm4xiRNBdzEOhA=";
+    # `openvpn3-core` is a submodule.
+    # TODO: make it into a separate package
+    fetchSubmodules = true;
+  };
+
+  patches = [
+    ./patches/0001-customizable-asio-path.patch
+    ./patches/0002-customizable-installation-paths.patch
+  ];
+
+  postPatch = ''
+    echo '#define OPENVPN_VERSION "3.git:unknown:unknown"' > ./src/build-version.h
+    echo '#define PACKAGE_GUIVERSION "${builtins.replaceStrings ["_"] [":"] version}_{unknown}__s"' >> ./src/build-version.h
+    echo '#define PACKAGE_NAME "openvpn3-linux"' >> ./src/build-version.h
```

Comment:
We set those headers in advance so that Meson doesn't try to set them by itself, right?

Reply:
Meson seems to rely on existence of `.git` directory to lazily generate this header file.

Since keeping dot-git is an anti-pattern for `fetchFromGitHub` (sometimes causing hash invalidation), I went with manually generating the header.

Reply:
Just pushed a one-liner ammend-commit 641d1d9b0abbb45865cfc9d974b66f0bbbe4daa9 removing redundant braces around `unknown`